string senderType
int32[] msgData
